app/lib/reports/pages/reports.dart

```diff
@@ -1,12 +1,203 @@
+              loading: () => Center(child: CircularProgressIndicator()),
+              loaded: (medications) => _buildReportsPage(medications, context),
+            ),
+          );
+        },
+      ),
+    );
+  }
+
+  Widget _buildReportsPage(
+    List<MedicationWithGuidelines> medications,
+    BuildContext context,
+  ) {
+    return CustomScrollView(slivers: [
+      SliverPersistentHeader(
+        delegate: SliverReportsHeaderDelegate(48, 96, 136),
+        floating: true,
+      ),
+      _buildMedicationsList(medications),
+    ]);
+  }
+
+  Widget _buildMedicationsList(List<MedicationWithGuidelines> medications) {
+    var filteredMedications = medications.map(filterUserGuidelines).toList();
+    filteredMedications = filteredMedications
+        .where(
+          (element) =>
+              element.guidelines.isNotEmpty &&
+              !_containsOnlyOkGuidelines(element.guidelines),
+        )
+        .toList();
+
+    return SliverList(
+      delegate: SliverChildBuilderDelegate((context, index) {
+        final med = filteredMedications[index];
+        return ReportCard(
+          warningLevel: _getWarningLevel(med.guidelines),
+          onTap: () {},
+          medicationName: med.name,
+          medicationDescription: med.description,
+        );
+      }, childCount: filteredMedications.length),
+    );
+  }
+
+  WarningLevel _getWarningLevel(List<Guideline> guidelines) {
+    for (final guideline in guidelines) {
+      if (guideline.warningLevel == WarningLevel.danger.name) {
+        return WarningLevel.danger;
+      }
+    }
+    return WarningLevel.warning;
+  }
+
+  bool _containsOnlyOkGuidelines(List<Guideline> guidelines) {
+    final warningLevels = guidelines.map((e) => e.warningLevel);
+    return warningLevels.every((warningLevel) {
+      return warningLevel == WarningLevel.ok.name;
+    });
+  }
+}
+
+class SliverReportsHeaderDelegate extends SliverPersistentHeaderDelegate {
+  const SliverReportsHeaderDelegate(
+    this.toolBarHeight,
+    this.closedHeight,
+    this.openHeight,
+  ) : super();
+
+  final double toolBarHeight;
+  final double closedHeight;
+  final double openHeight;
+
+  @override
+  double get maxExtent => toolBarHeight + openHeight;
+
+  @override
+  double get minExtent => toolBarHeight + closedHeight;
+
+  @override
+  bool shouldRebuild(SliverPersistentHeaderDelegate oldDelegate) => true;
+
+  @override
+  Widget build(
+    BuildContext context,
+    double shrinkOffset,
+    bool overlapsContent,
+  ) {
+    return Card(
+      shape: RoundedRectangleBorder(
+        borderRadius: BorderRadius.circular(16),
+      ),
+      color: PharmeTheme.secondaryColor,
+      clipBehavior: Clip.hardEdge,
+      child: Padding(
+        padding: const EdgeInsets.all(8),
+        child: Column(children: [
+          Text(
+            context.l10n.reports_page_disclaimer_title,
+            style:
+                PharmeTheme.textTheme.headline6!.copyWith(color: Colors.white),
```

Comment:
```suggestion
                PharmeTheme.textTheme.titleSmall!.copyWith(color: Colors.white),
```
headline1-6, subtitle1-2, body1-2, caption, button and overline are deprecated.

display, headline, title, body, label (suffixed with Large, Medium, Small respectively) are the new standards. You can see their interpretation [here](https://api.flutter.dev/flutter/material/TextTheme-class.html)